Use serialization for WebCore::ExceptionCode

mikhailramalho opened the pull request:
#### 87ce4ccda0c65191a9b808891842b4eee7eaaeaf
<pre>
Use serialization for WebCore::ExceptionCode
<a href="https://bugs.webkit.org/show_bug.cgi?id=264453">https://bugs.webkit.org/show_bug.cgi?id=264453</a>

Reviewed by Chris Dumez.

OutOfMemoryError was missing from EnumTraits, so it was added to the enum class
in WebCoreArgumentCoders.serialization.in

* Source/WebCore/dom/ExceptionCode.h:
(): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/270558@main">https://commits.webkit.org/270558@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6d56ddc7364a22c3bedc5e2030ad055e6217f50a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25749 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4356 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27033 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27850 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23586 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26034 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6113 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1792 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23702 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25999 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3269 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22193 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28430 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2894 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23147 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29220 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23504 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23517 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27083 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2910 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1144 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4293 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3362 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3299 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3226 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (changes requested):
There's a lot of noise in this PR distracting us from the relevant change, which is only in three files (`ExceptionCode.{h|cpp}` and `WebCoreArgumentCoders.specialization.in`). I think this would be much easier to review if you could kindly break it down to two patches.

1. `WebCore::ExceptionCode` should be an `enum class`
   - This will be a mechanical patch that will get rid of much of the noise from the real change in this patch.
2. Use serialization for `WebCore::ExceptionCode` (this patch)
   - This will only have the changes made in `WebCoreArgumentCoders.serialization.in` and the removal of the `EnumTraits<>` specialization.

cdumez commented on Source/WebCore/Modules/ShapeDetection/BarcodeDetector.cpp:
```diff
@@ -47,19 +47,19 @@ ExceptionOr<Ref<BarcodeDetector>> BarcodeDetector::create(ScriptExecutionContext
         const auto* page = document.page();
         if (!page)
-            return Exception { AbortError };
+            return Exception { WebCore::ExceptionCode::AbortError };
```
We don’t need WebCore:: since we are in WebCore. Same comment applies elsewhere.

mikhailramalho replied:
Changed in https://github.com/WebKit/WebKit/pull/20237. Only a couple of places needed the `WebCore::` prefix, e.g., `NetworkProcess/storage/FileSystemStorageError.h`

cdumez reviewed (approved):
LGTM but please fix updated files in the commit log message.

mikhailramalho committed:
Use serialization for WebCore::ExceptionCode
https://bugs.webkit.org/show_bug.cgi?id=264453

Reviewed by Chris Dumez.

OutOfMemoryError was missing from EnumTraits, so it was added to the enum class
in WebCoreArgumentCoders.serialization.in

* Source/WebCore/dom/ExceptionCode.h:
(): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/270558@main
